maestro/kubecontrol/monitor.go

```diff
@@ -0,0 +1,234 @@
+		svc.logger.Error("error on get client", zap.Error(err))
+		return nil, err
+	}
+	pods, err := clientSet.CoreV1().Pods(namespace).List(ctx, k8smetav1.ListOptions{})
+	return pods.Items, err
+}
+
+func (svc *monitorService) monitorSinks(ctx context.Context) {
+	runningCollectors, err := svc.getRunningPods(ctx)
+	if err != nil {
+		svc.logger.Error("error getting running pods on namespace", zap.Error(err))
+		return
+	}
+	if len(runningCollectors) == 0 {
+		svc.logger.Info("skipping, no running collectors")
+		return
+	}
+	sinksRes, err := svc.sinksClient.RetrieveSinks(ctx, &sinkspb.SinksFilterReq{OtelEnabled: "enabled"})
+
+	if err != nil {
+		svc.logger.Error("error collecting sinks", zap.Error(err))
+		return
+	}
+	svc.logger.Info("reading logs from collectors", zap.Int("collectors_length", len(sinksRes.Sinks)))
+	for _, sink := range sinksRes.Sinks {
+		var sinkCollector *k8scorev1.Pod
+		for _, collector := range runningCollectors {
+			if strings.Contains(collector.Name, sink.Id) {
+				sinkCollector = &collector
+				break
+			}
+		}
+		if sinkCollector == nil {
+			svc.logger.Warn("collector not found for sink, skipping", zap.String("sinkID", sink.Id))
+			continue
+		}
+		var data maestroconfig.SinkData
+		if err := json.Unmarshal(sink.Config, &data); err != nil {
+			svc.logger.Warn("failed to unmarshal sink, skipping", zap.String("sink-id", sink.Id))
+			continue
+		}
+		data.SinkID = sink.Id
+		data.OwnerID = sink.OwnerID
+		data.LastRemoteWrite = time.Now()
+		logs, err := svc.getPodLogs(ctx, *sinkCollector)
+		if err != nil {
+			svc.logger.Error("error on getting logs, skipping", zap.Error(err))
+			continue
+		}
+		status, logsErr := svc.analyzeLogs(logs)
+		if status == "fail" {
+			svc.logger.Error("error during analyze logs", zap.Error(logsErr))
+		}
+		if data.State.String() != status {
+			if err != nil {
+				svc.logger.Info("updating status", zap.Any("before", sink.GetState()), zap.String("new status", status), zap.String("error_message (opt)", err.Error()))
+			} else {
+				svc.logger.Info("updating status", zap.Any("before", sink.GetState()), zap.String("new status", status))
+			}
+			keyPrefix := "sinker_key"
+			skey := fmt.Sprintf("%s-%s:%s", keyPrefix, data.OwnerID, data.SinkID)
+			err := data.State.SetFromString(status)
+			if err != nil {
+				svc.logger.Error("error during analyze logs", zap.Error(err))
+			}
+			if logsErr != nil {
+				data.Msg = logsErr.Error()
+			}
+			if err := svc.redisClient.Del(ctx, skey).Err(); err != nil {
+				svc.logger.Error("error during analyze logs", zap.Error(err))
+			}
+			byteSink, err := json.Marshal(data)
+			if err != nil {
+				svc.logger.Error("error during analyze logs", zap.Error(err))
+			}
+			if err = svc.redisClient.Set(context.Background(), skey, byteSink, 0).Err(); err != nil {
+				svc.logger.Error("error during analyze logs", zap.Error(err))
+			}
+		}
+		return
+	}
+
+}
+
+// analyzeLogs, will check for errors in exporter, and will return as follows
```

Comment:
can you clarify the docs here - _what_ is the format of the log it is analyzing, open telemetry collector? i.e., a change in which software might cause us to need to make changes here?

Reply:
I am not happy with this piece of code, I am trying to figure out a better plan for this method.
Otel uses the 
```
import (
    "log"
)
```
package to log, I will open a ticket and try to standardize to something that we can log as json, so we can read these log entries better.

Reply:
https://pkg.go.dev/log